### PR TITLE
[dem] A histogram with the number of bonds is printed

### PR DESCRIPTION
**Pull request by maceligueta**
It is written in the graphs folder.

### CODE REVIEW

**Review by salvalatorre — DISMISSED** [review]
More than OK in my view...

File: applications/DEMApplication/custom_utilities/pre_utilities.h [hunk 1]
```diff
@@ -83,6 +83,26 @@ class PreUtilities
         p_properties->SetValue(CLUSTER_INFORMATION, cl_info);
     }
 
+    void PrintNumberOfNeighboursHistogram(ModelPart& rSpheresModelPart, std::string const& filename) {
```
Comment (philbucher) [inline applications/DEMApplication/custom_utilities/pre_utilities.h]:
```suggestion
    void PrintNumberOfNeighboursHistogram(const ModelPart& rSpheresModelPart, std::string const& filename) {
```

**Review by salvalatorre — DISMISSED** [review]
Even better now!

File: applications/DEMApplication/custom_utilities/pre_utilities.h [hunk 2]
```diff
@@ -83,6 +83,38 @@ class PreUtilities
...
+        for(int i=0; i<(int)number_of_spheres_with_i_neighbours.size(); i++) {number_of_spheres_with_i_neighbours[i] = 0;}
+
+        const ElementsArrayType& pElements = rSpheresModelPart.GetCommunicator().LocalMesh().Elements();
+        ElementsArrayType::ptr_const_iterator begin = pElements.ptr_begin();
+
+        for(int i=0; i<(int)pElements.size(); i++) {
+            ElementsArrayType::ptr_const_iterator it = begin + i;
+            const Element& el = **it;
+            const SphericContinuumParticle* p_cont_sphere = dynamic_cast<const SphericContinuumParticle*>(&el);
+            if(p_cont_sphere != NULL) {
```
Comment (philbucher) [inline applications/DEMApplication/custom_utilities/pre_utilities.h]:
```suggestion
            if(p_cont_sphere) {
```

File: applications/DEMApplication/custom_utilities/pre_utilities.h [hunk 3]
```diff
@@ -83,6 +83,38 @@ class PreUtilities
...
+                const SphericParticle* p_sphere = dynamic_cast<const SphericParticle*>(&el);
+                unsigned int size = p_sphere->mNeighbourElements.size();
+                if(size > number_of_spheres_with_i_neighbours.size() - 1) size = number_of_spheres_with_i_neighbours.size() - 1;
+                number_of_spheres_with_i_neighbours[size] += 1;
+            }
+        }
+        std::ofstream outputfile(filename, std::ios_base::out | std::ios_base::app);
+        outputfile << "number_of_neighbours   percentage_of_spheres_with_that_number_of_neighbours    number_of_spheres_with_that_number_of_neighbours\n";
+        for(int i=0; i<(int)number_of_spheres_with_i_neighbours.size(); i++) {
+            const double percentage = (double)(number_of_spheres_with_i_neighbours[i]) / rSpheresModelPart.NumberOfElements(0) * 100;
```
Comment (philbucher) [inline applications/DEMApplication/custom_utilities/pre_utilities.h]:
```suggestion
            const double percentage = (double)(number_of_spheres_with_i_neighbours[i]) / double(rSpheresModelPart.NumberOfElements(0)) * 100.0;
```

doesn't this also need to be casted?

File: applications/DEMApplication/custom_utilities/pre_utilities.h [hunk 4]
```diff
@@ -83,6 +83,38 @@ class PreUtilities
...
+                const SphericParticle* p_sphere = dynamic_cast<const SphericParticle*>(&el);
+                unsigned int size = p_sphere->mNeighbourElements.size();
+                if(size > number_of_spheres_with_i_neighbours.size() - 1) size = number_of_spheres_with_i_neighbours.size() - 1;
+                number_of_spheres_with_i_neighbours[size] += 1;
+            }
+        }
+        std::ofstream outputfile(filename, std::ios_base::out | std::ios_base::app);
+        outputfile << "number_of_neighbours   percentage_of_spheres_with_that_number_of_neighbours    number_of_spheres_with_that_number_of_neighbours\n";
+        for(int i=0; i<(int)number_of_spheres_with_i_neighbours.size(); i++) {
+            const double percentage = (double)(number_of_spheres_with_i_neighbours[i]) / double(rSpheresModelPart.NumberOfElements(0)) * 100.0;
```
Comment (philbucher) [inline applications/DEMApplication/custom_utilities/pre_utilities.h]:
```suggestion
            const double percentage = (double)(number_of_spheres_with_i_neighbours[i]) / (double)(rSpheresModelPart.NumberOfElements(0)) * 100.0;
```

File: applications/DEMApplication/custom_utilities/pre_utilities.h (hunk 3 above)
Comment (maceligueta) [reply applications/DEMApplication/custom_utilities/pre_utilities.h]: In principle it is not necessary. As long as one of the numbers is a double the whole operation is double. But I think your version is cleaner.

**Review by salvalatorre — APPROVED** [review]
Now the PR is simply awesome!

**Pull request merged**